lib/dgram.js

```diff
@@ -272,6 +267,65 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+function fixBuffer(buffer) {
```

Comment:
nit: function name is slightly incomplete? maybe something like `fixAndSumBuffers()`. this is nothing serious, and not a blocker.